### DISCUSSION TITLE
sending data safety

### DISCUSSION THREAD

**Discussion by lab6net** [ant-media/Ant-Media-Server#6692] [General]
Hello everyone

I'am quite new to ant server technology. Been using it for last couple weeks to make some simple streaming service with chat feature. For chat i wanted to use data channel that is available in ant server (streaming through webrtc).

Currently in application settings i can enable/disable data transfer, or allow only to send data by publisher, or by all. I would like to have functionality where everybody can receive the data (that is accomplished by setting data channel to "Publisher and all Players") but nobody can send the data except through rest api. This way i can use my own app as a man in the middle to control those messages and block some stuff (im not only sending chat messages but also various events, and i dont want somebody to send those to everybody just py preparing specific message).

So what i have currently is:
Client are receiving data by standard notification in the callback "data_received", then if they want to send message, they are calling ajax request that goes to my app. My app checks if its correct then makes api call to ant rest api to broadcast that message. The only thing im missing is to block connected users to send anything through data channel without verification.

**Comment by lastpeony** [comment]
If your application has complex chat functionality and event management, I recommend using WebSockets with a library like Socket.IO for those features, while relying on Ant Media Server solely for audio and video streaming. This approach provides greater flexibility and allows you to handle chat and events more efficiently.
This is just my opinion someone from community might advice something else.

**Reply by rahul78275** [reply]
I agree with you @lastpeony , Socket.io is my first preference. Very flexible and easy to use.

**Reply by lab6net** [reply]
I'am aware that i can use socket.io. It is just very convinient that i could utilize data channel, else what is this feature for ?

**Reply by burak-58** [reply]
If your users are already connected with WebRTC, then using WebRTC instead of socket.io is meaningful.

@lab6net you want to prevent users send data channel messages throug JS SDK, right?

**Reply by lab6net** [reply]
> If your users are already connected with WebRTC, then using WebRTC instead of socket.io is meaningful.
> 
> @lab6net you want to prevent users send data channel messages throug JS SDK, right?

Yes, i want them to only receive it through data channel, sending will be done through normal http requests so my app can control it.
